joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/astcreation/AstForControlStructuresCreator.scala

```diff
@@ -114,7 +116,22 @@ trait AstForControlStructuresCreator(implicit withSchemaValidation: ValidationMo
 
     val catches = stmt.catches.map { catchStmt =>
       val catchNode = controlStructureNode(catchStmt, ControlStructureTypes.CATCH, "catch")
-      Ast(catchNode).withChild(astForCatchStmt(catchStmt))
+
+      val local = catchStmt.variable.collectFirst { case variable @ PhpVariable(name: PhpNameExpr, _) =>
+        val local = localNode(variable, name.name, name.name, Defines.Any)
+        scope.addToScope(name.name, local) match {
+          case PhpScopeElement(node: NewBlock) => diffGraph.addEdge(node, local, EdgeTypes.AST)
+          case _                               =>
+        }
+        local.dynamicTypeHintFullName(catchStmt.types.map(_.name))
+        Ast(local)
+      }
+
+      val catchAst = Ast(catchNode).withChild(astForCatchStmt(catchStmt))
```

Comment:
I don't think you end up pushing the block scope for the catch body here do you? i.e., `scope.pushNewScope`